### ISSUE TITLE
Want to see the CNN Filter (image format)

### ISSUE THREAD

**Issue by UdayKiranPadhy** [tensorflow/tensorflow#41723]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using):
2.2.0
- Are you willing to contribute it (Yes/No):
Yes

**Describe the feature and the current behavior/state.**
There is no such feature for what filter did the model came up while training is completed. Please make a function for it. 

**Will this change the current api? How?**
The API will become more Awosome than before it was. Present every one is getting accuracy on just predefined test and train cases , but after this feature people will know what their model is predecting in the given image

**Who will benefit with this feature?**
Especailly people who are learning like me rather than the people who are just run behind accuracy
**Any Other info.**
Please If u want any help , do tell me , Big fan of Google's Code Works . And also make a function for it like model.show_filter_con2d() ;

**Comment by ravikyram** [comment]
@UdayKiranPadhy 

Do you have any use case that requires the feature you are interested in? Please feel free to submit a PR if you have use cases that supports that feature.Thanks!

**Comment by UdayKiranPadhy** [comment]
Do you have any use case that requires the feature you are interested in?
For example, When i am training a model on a dataset of cats and dogs . I want to know what my model learn from those images. Did it successfully recognize its eyes and ears or not ? Yeah it may be tested using test data set but i want to know on bases on what features the model is predicting the image is cat or a dog

Please feel free to submit a PR if you have use cases that supports that feature.Thanks!
What is PR and how to submit it . Can u Please provide me a link for it .

Thank You , Have a nice day

**Comment by ravikyram** [comment]
@UdayKiranPadhy 

Can you please refer the [link](https://www.tensorflow.org/community/contribute/code) to submit PR.Thanks!

**Comment by UdayKiranPadhy** [comment]
Hey i dont know how to code , i only suggested a feature for tensorflow library , if i knew to code i would implement my self why will i tell to tenserflow , The link which u send me is telling me to contribute via code .

**Comment by jvishnuvardhan** [comment]
@UdayKiranPadhy There are many tutorials on how to plot weights. For example, Can you please follow this [resource](https://machinelearningmastery.com/how-to-visualize-filters-and-feature-maps-in-convolutional-neural-networks/). 

Please feel free to close the issue if this resolved the issue for you. Thanks!

**Comment by rmothukuru** [comment]
@UdayKiranPadhy,
In addition to the resource above, which shows how to visualize Feature Maps and Filters,  Please find [this article](https://www.pyimagesearch.com/2020/03/09/grad-cam-visualize-class-activation-maps-with-keras-tensorflow-and-deep-learning/) which demonstrates how to Visualize Class Activation Maps i.e., what features of an Image contributes to the prediction corresponding to its class. Thanks!
